Parameter Documentation checker only works if at least one parameter is documented properly

<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this, make sure that the issue doesn't already exist
  or if it is not closed.

  Is your issue fixed on the preview release?: pip install pylint astroid --pre -U

-->

### Steps to reproduce
1. Enable `pylint.extensions.docparams`
2. Any of the following functions will be rated 10/10
```python
def foobar1(arg1, arg2):
    """function foobar ...
    """
    print(arg1, arg2)

def foobar2(arg1, arg2):
    """function foobar ...

    Parameters
    ----------
    """
    print(arg1, arg2)

def foobar3(arg1, arg2):
    """function foobar ...

    Parameters
    ----------
    arg1: int
    """
    print(arg1, arg2)
```
3. Run pylint

### Current behavior
The checker doesn't raise any warnings related to the parameters on the documentation
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior
Something similar to this:

```python
def foobar4(arg1, arg2):
    """function foobar ...

    Parameters
    ----------
    arg1: int
        description
    """
    print(arg1, arg2)
```

That will return the warnings for any missing parameter:
```
************* Module pylint_config
pylint_config.py:1:0: W9015: "arg2" missing in parameter documentation (missing-param-doc)
pylint_config.py:1:0: W9016: "arg2" missing in parameter type documentation (missing-type-doc)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```


### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.6.4 (default, Jul 22 2019, 12:18:30)
[GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)]
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -425,3 +425,5 @@ contributors:
 * Giuseppe Valente: contributor
 
 * Takashi Hirashima: contributor
+
+* Joffrey Mander: contributor

--- a/ChangeLog
+++ b/ChangeLog
@@ -15,6 +15,10 @@ What's New in Pylint 2.6.1?
 ===========================
 Release date: TBA
 
+* Fix linter multiprocessing pool shutdown (triggered warnings when runned in parallels with other pytest plugins)
+
+  Closes #3779
+
 * Fix a false-positive emission of `no-self-use` and `unused-argument` for methods
   of generic structural types (`Protocol[T]`)
 
@@ -23,7 +27,7 @@ Release date: TBA
 * Fix bug that lead to duplicate messages when using ``--jobs 2`` or more.
 
   Close #3584
- 
+
 * Adds option ``check-protected-access-in-special-methods`` in the ClassChecker to activate/deactivate
   ``protected-access`` message emission for single underscore prefixed attribute in special methods.
 

--- a/doc/whatsnew/2.6.rst
+++ b/doc/whatsnew/2.6.rst
@@ -29,6 +29,8 @@ New checkers
 Other Changes
 =============
 
+* Fix linter multiprocessing pool shutdown which triggered warnings when runned in parallels with other pytest plugins.
+
 * Enums are now required to be named in UPPER_CASE by ``invalid-name``.
 
 * Fix bug that lead to duplicate messages when using ``--jobs 2`` or more.

--- a/pylint/lint/parallel.py
+++ b/pylint/lint/parallel.py
@@ -91,15 +91,16 @@ def check_parallel(linter, jobs, files, arguments=None):
     # is identical to the linter object here. This is requred so that
     # a custom PyLinter object can be used.
     initializer = functools.partial(_worker_initialize, arguments=arguments)
-    with multiprocessing.Pool(jobs, initializer=initializer, initargs=[linter]) as pool:
-        # ..and now when the workers have inherited the linter, the actual reporter
-        # can be set back here on the parent process so that results get stored into
-        # correct reporter
-        linter.set_reporter(original_reporter)
-        linter.open()
+    pool = multiprocessing.Pool(jobs, initializer=initializer, initargs=[linter])
+    # ..and now when the workers have inherited the linter, the actual reporter
+    # can be set back here on the parent process so that results get stored into
+    # correct reporter
+    linter.set_reporter(original_reporter)
+    linter.open()
 
-        all_stats = []
+    all_stats = []
 
+    try:
         for (
             module,
             file_path,
@@ -116,6 +117,9 @@ def check_parallel(linter, jobs, files, arguments=None):
 
             all_stats.append(stats)
             linter.msg_status |= msg_status
+    finally:
+        pool.close()
+        pool.join()
 
     linter.stats = _merge_stats(all_stats)
 

